content/hi/docs/concepts/architecture/garbage-collection.md

```diff
@@ -0,0 +1,183 @@
+
+फोरग्राउंड कैस्केडिंग विलोपन में, आप जिस स्वामी वस्तु को हटा रहे हैं, वह पहले प्रवेश करती है
+a *हटाना जारी है* स्थिति। इस अवस्था में निम्नलिखित होता है
+स्वामी वस्तु:
+
+  * Kubernetes API सर्वर ऑब्जेक्ट का `metadata.deletionTimestamp` सेट करता है
+    फ़ील्ड उस समय तक जब ऑब्जेक्ट को हटाने के लिए चिह्नित किया गया था।
+  * Kubernetes API सर्वर `metadata. finalizers` फ़ील्ड को भी सेट करता है
+    `अग्रभूमि हटाना`।
+  * ऑब्जेक्ट कुबेरनेट्स एपीआई के माध्यम से विलोपन तक दृश्यमान रहता है
+    प्रक्रिया पूरी हो गई है।
+
+स्वामी वस्तु के विलोपन की प्रगति की स्थिति में प्रवेश करने के बाद, नियंत्रक
+आश्रितों को हटाता है। सभी आश्रित वस्तुओं को हटाने के बाद, नियंत्रक
+स्वामी वस्तु को हटाता है। इस बिंदु पर, वस्तु अब में दिखाई नहीं दे रही है
+कुबेरनेट्स एपीआई।
+
+अग्रभूमि कैस्केडिंग विलोपन के दौरान, एकमात्र आश्रित जो स्वामी को ब्लॉक करते हैं
+विलोपन वे हैं जिनके पास `ownerReference.blockOwnerDeletion=true` फ़ील्ड है।
+[अग्रभूमि कैस्केडिंग विलोपन का उपयोग करें](/docs/tasks/administer-cluster/use-cascading-deletion/#use-foreground-cascading-deletion) देखें
+अधिक जानने के लिए।
+
+### बैकग्राउंड कैस्केडिंग डिलीट {#बैकग्राउंड-डिलीशन}
+
+बैकग्राउंड कैस्केडिंग विलोपन में, Kubernetes API सर्वर स्वामी को हटा देता है
+वस्तु तुरंत और नियंत्रक आश्रित वस्तुओं को साफ करता है
+पृष्ठभूमि। डिफ़ॉल्ट रूप से, कुबेरनेट पृष्ठभूमि कैस्केडिंग विलोपन का उपयोग करता है जब तक कि
+आप मैन्युअल रूप से अग्रभूमि विलोपन का उपयोग करते हैं या आश्रित वस्तुओं को अनाथ करना चुनते हैं।
+
+देखें [बैकग्राउंड कैस्केडिंग डिलीट का उपयोग करें](/docs/tasks/administer-cluster/use-cascading-deletion/#use-background-cascading-deletion) अधिक जानने के लिए।
+
+### अनाथ आश्रित
+
+जब कुबेरनेट्स किसी स्वामी वस्तु को हटाते हैं, तो पीछे रह गए आश्रितों को बुलाया जाता है
+*अनाथ* वस्तुएं। डिफ़ॉल्ट रूप से, कुबेरनेट्स आश्रित वस्तुओं को हटा देता है। जानने के लिए कैसे
+इस व्यवहार को ओवरराइड करने के लिए, [स्वामी वस्तुओं और अनाथ आश्रितों को हटाएं] (/docs/tasks/administer-cluster/use-cascading-deletion/#set-orphan-deletion-policy) देखें।
+
+## अप्रयुक्त कंटेनरों और छवियों का कचरा संग्रह {#containers-images}
+
+{{<glossary_tooltip text="kubelet" term_id="kubelet">}} कचरा करता है
+प्रत्येक पांच मिनट में अप्रयुक्त छवियों पर संग्रह और अप्रयुक्त कंटेनरों पर प्रत्येक
+मिनट। आपको बाहरी कचरा संग्रह उपकरण का उपयोग करने से बचना चाहिए, क्योंकि ये कर सकते हैं
+क्यूबलेट व्यवहार को तोड़ें और मौजूद कंटेनरों को हटा दें।
+
+अप्रयुक्त कंटेनर और छवि कचरा संग्रह के लिए विकल्पों को कॉन्फ़िगर करने के लिए, ट्यून करें
+[कॉन्फ़िगरेशन फ़ाइल](/docs/tasks/administer-cluster/kubelet-config-file/) का उपयोग करके क्यूबलेट
+और कचरा संग्रहण से संबंधित मापदंडों को उपयोग करके बदलें
+[`KubletConfiguration`](/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubletConfiguration)
+resource type.
+
+### कंटेनर छवि जीवनचक्र
+
+कुबेरनेट्स अपने *छवि प्रबंधक* के माध्यम से सभी छवियों के जीवनचक्र का प्रबंधन करता है,
+जो क्यूबलेट का हिस्सा है, के सहयोग से
+{{< glosary_tooltip text="cadvisor" term_id="cadvisor" >}}. क्यूबलेट
```

Comment:
Typo in the `glossary_tooltip` shortcode name. Please consider localizing the "text" parameter passed to the shortcode to Hindi.